src/Grpc/JsonTranscoding/src/Microsoft.AspNetCore.Grpc.JsonTranscoding/Internal/DescriptorRegistry.cs

```diff
@@ -45,29 +57,28 @@ private void AddFileDescriptorsRecursive(FileDescriptor fileDescriptor)
         }
     }
 
-    private void AddNestedEnumDescriptorsRecursive(MessageDescriptor messageDescriptor)
+    private void AddDescriptorsRecursive(MessageDescriptor messageDescriptor)
     {
+        // Type is null for map entry message types. Just skip adding them.
+        if (messageDescriptor.ClrType != null)
+        {
+            _typeDescriptorMap[messageDescriptor.ClrType] = messageDescriptor;
+        }
+
         foreach (var enumDescriptor in messageDescriptor.EnumTypes)
         {
-            _enumDescriptors.Add(enumDescriptor);
+            _typeDescriptorMap[enumDescriptor.ClrType] = enumDescriptor;
```

Comment:
Is there some reason to believe this won't clobber an existing value?  Or maybe clobbering does happen and it's important?

Reply:
It shouldn't conflict